### ISSUE TITLE
Add `weight_decay_filter` and `lars_adaptation_filter` to LARS

### ISSUE THREAD

**Issue by turian** [Lightning-AI/lightning-flash#1432]
## 🚀 Feature

Add `weight_decay_filter` and `lars_adaptation_filter` to LARS

### Motivation

weight decay typically shouldn't be applied to BatchNorm. See [fast.ai](https://www.fast.ai/2018/08/10/fastai-diu-imagenet/) and this [pytorch discuss thread](https://discuss.pytorch.org/t/weight-decay-in-the-optimizers-is-a-bad-idea-especially-with-batchnorm/16994).

The facebook [vicreg code](https://github.com/facebookresearch/vicreg/blob/main/main_vicreg.py#L245) has parameters `weight_decay_filter` and `lars_adaptation_filter` which they set to True for any parameter that has ndim 1.

### Pitch

There should be a simple way to disable weight decay and LARS adaptation on ndim==1 parameters.

### Alternatives

Port Facebook LARS code and use it instead of lightning flash LARS code.

**Comment by krshrimali** [comment]
Hi, @turian - Thank you for creating the issue. Just to let you know, I have this on my list to take a look at, and I'll try to get back by this weekend. A bit occupied, apologies for the delay.

**Comment by krshrimali** [comment]
Hi, @turian - Thank you for giving the context, I went through the discussion on the PyTorch forum. I think it's fair to give an option to the user to disable this based on the condition (`ndim == 1`). Would you like to create a PR to add this? If not, I'll be able to take a look, hopefully soon. Thank you! ⚡

**Comment by turian** [comment]
@krshrimali I am not sure that I would able to create a PR that covers all corner cases. :(

**Comment by krshrimali** [comment]
> @krshrimali I am not sure that I would able to create a PR that covers all corner cases. :(

No worries at all! I will try to take a look, we are working towards a release tomorrow, so I will need some time but I have added this to my list. Thank you again!!

**Comment by turian** [comment]
@krshrimali Great! I am following this issue.

**Comment by krshrimali** [comment]
> @krshrimali Great! I am following this issue.

I'll try to pick this up over the coming weekend. 🤞🏻 Thanks for your patience, @turian 🚀

**Comment by turian** [comment]
@krshrimali Thanks! And I am happy to help with code review if you tag me in the PR

**Comment by krshrimali** [comment]
> @krshrimali Thanks! And I am happy to help with code review if you tag me in the PR

Thanks! I'll make sure to request your review :)
